frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -887,6 +887,28 @@ export default ({
         }
 
         Vue.set(this.messageState, 'contract', state)
+
+        // After message initialization, scroll to the determined initial scroll position.
+        // (eg. particular message queried by 'mhash', remembered chatroom scroll position or the most recent message)
+        if (this.ephemeral.scrollHashOnInitialLoad) {
+          const scrollingToSpecificMessage = this.$route.query?.mhash === this.ephemeral.scrollHashOnInitialLoad
+          this.$nextTick(() => {
+            this.updateScroll(
+              this.ephemeral.scrollHashOnInitialLoad,
+              // NOTE: we do want the 'c-focused' animation if there is a scroll-to-message query.
+              scrollingToSpecificMessage
+            )
+            // NOTE: delete mhash in the query after scroll and highlight the message with mhash
+            if (scrollingToSpecificMessage) {
+              const newQuery = { ...this.$route.query }
+              delete newQuery.mhash
+              this.$router.replace({ query: newQuery })
+            }
+
+            // This ensures that scroll-to-specific-message (or to the latest message) happens one time after the initial chatroom load.
+            this.ephemeral.scrollHashOnInitialLoad = null
+          })
+        }
```

Comment:
From more thorough investigation, this appears to be a better place to execute the `initial-scroll` logic. From my observation, executing this logic from the original place (L1092 below) seemed premature especially when the chatroom has many image attachments. This leads to a situation like below for example,

[1] Some messages are rendered.
[2] initial-scroll logic is executed.
[3] Images that are rendered later causes the scroll-height of the chatroom to increase.
[4] This leads to a sudden jump of the scroll position.

After moving this logic here, I have not observed this happening again.